### DISCUSSION TITLE
Accessing global data

### DISCUSSION THREAD

**Discussion by AL89** [DHI/mikeio1d#58] [Q&A] [answered]
Hi there

So, I would like to get the global values in the form of `WaterVolume` for my sewer system.

I have a Network HD Result file (`result_file.res1d`) with all quantities for nodes, reaches (links) and system volume (global data).
How do I access this?

    from mikeio1d.res1d import Res1D

    mike_result_file = Res1D(result_file.res1d)
    mike_result_file.global_data

The above gives me:

    {'WaterVolume': <DHI.Mike1D.ResultDataAccess.DataItem object at 0x000001E04A65ACC0>}

So, I can see it is there, but how do I access the data/values?

Thanks in advance!

**Comment by AL89** [comment]
I feel like a complete moron! 

I just added this in the `import`

    from mikeio1d.res1d import Res1D, QueryDataGlobal

The `QueryDataGlobal` gives an opportunity to make a query for global data, obviously, like `WaterVolume` and `TimeStep` data.

So, I can read the query in my result file, doing this:

    waterVol_query = QueryDataGlobal('WaterVolume')
    mike_result_file.read(waterVol_query)

Thus, it returns a `pandas DataFrame`.

Sorry bothering you all!

**Reply by ryan-kipawa** [reply, marked as answer]
Hi there,

Glad that you found a solution. We actually really appreciate posts like this to help make MIKEIO 1D more intuitive for users, so thanks!

In the latest version (v0.4.0), we tried to make this a bit easier. You should now be able to accomplish the above by:

```
from mikeio1d import Res1D
res = Res1D('my_result_file.res1d')
df = res.result_network.global_data.WaterVolume.read()
```

You may also notice that there's autocompletion for the quantity types if you're using an IDE like vscode.

If you have any suggestions, feel free to make them on these boards :)

Cheers,
Ryan

**Reply by AL89** [reply]
Hi @rywm-dhi 

I guess your solution was the one I needed. A very intuitive way to access the global data as well as other data. Didn't know that shortcut (`result_network`) existed. Thanks!
